homeassistant/util/yaml.py

```diff
@@ -144,7 +144,7 @@ def _find_files(directory: str, pattern: str) -> Iterator[str]:
     """Recursively load files in a directory."""
     for root, dirs, files in os.walk(directory, topdown=True):
         dirs[:] = [d for d in dirs if _is_file_valid(d)]
-        for basename in files:
+        for basename in sorted(files):
```

Comment:
This will impact ALL commands that are using this function. This one-word patch needs to have a better description which commands are being impacted, how it was before and how it is now. 

I am not familiar with this code. I would love some other developers experienced with the YAML includes to comment if this would be acceptable and make sure we correctly judge the impact.

Reply:
Thanks Balloob - 

Yes, it's correct that all four include_dir-type commands (described [here](https://www.home-assistant.io/docs/configuration/splitting_configuration/)) use this function to get the list of files.  They are the only things that use it.   Two of them return dictionaries, so the order is not visible to the end user anyway.

At present, the order in which files are returned by this `_find_files()` function is undefined, so no existing functionality should depend on it, or be adversely affected by imposing an ordering.  

All we are doing here is making a guarantee that was not there before, that for the two functions which do return lists, their order will now be predictable for any given set of input files.    

I have only commented on and added documentation for the simple `!include_dir_list`.  It is also the case that the order of the output of `!include_dir_merge_list` will be affected, but I didn't bother documenting it.  Nobody can be relying on the existing order, but we could also document the fact that _this_ ordering will now be predictable too - I thought it might be more complication than was warranted.

Reply:
I'm not an expert on yaml, but I agree with quentin, this appears to be super safe and it goes from not providing a guaranteed order to providing one.